utilities/src/errors.rs

```diff
@@ -15,7 +15,14 @@
 
 use colored::Colorize;
 
-use std::{borrow::Borrow, fmt::Display};
+use std::{any::Any, backtrace::Backtrace, borrow::Borrow, cell::Cell, fmt::Display, panic};
+
+thread_local! {
+/// The message backtrace of the last panic on this thread (if any).
+///
+/// We store this information here instead of directly processing it in a panic hook, because panic hooks are global whereas this can be processed on a per-thread basis.
+/// For example, one thread may execute a program where panics should *not* cause the entire process to terminate, while in another thread there is a panic due to a bug.
```

Comment:
This looks like quite a crucial and useful change which will save us a lot of headache in the future, thank you for the proposal. Same question as before, I'm curious to hear about prior art in the Rust world

Reply:
There is ongoing work to atomically update the panic handler (in [this RFC](github.com/rust-lang/rust/issues/92649)).

However, that still does not allow setting a handler per thread. For that, the common suggestion I found is essentially what I implemented.